src/ConcordiumNetSdk/Types/AccountAddress.cs

```diff
@@ -79,31 +101,35 @@ public static bool IsValid(string? addressAsBase58String)
         }
     }
 
-    public override string ToString()
+    /// <summary>
+    /// Converts the account address to its corresponding protocol buffer message instance.
+    /// </summary>
+    public Concordium.V2.AccountAddress ToProto()
     {
-        return _formatted;
+        return new Concordium.V2.AccountAddress()
+        {
+            Value = Google.Protobuf.ByteString.CopyFrom(this._value)
+        };
     }
 
     public override bool Equals(object? obj)
     {
-        if (ReferenceEquals(null, obj)) return false;
-        if (ReferenceEquals(this, obj)) return true;
-        if (obj.GetType() != GetType()) return false;
-        return AsString == ((AccountAddress) obj).AsString;
-    }
-
-    public override int GetHashCode()
-    {
-        return AsString.GetHashCode();
+        if (ReferenceEquals(null, obj))
+            return false;
+        if (ReferenceEquals(this, obj))
+            return true;
+        if (obj.GetType() != GetType())
+            return false;
+        return ToString() == ((AccountAddress)obj).ToString();
     }
 
-    public static bool operator ==(AccountAddress? left, AccountAddress? right)
+    public bool Equals(AccountAddress? accountAddress)
     {
-        return Equals(left, right);
+        return this.Equals((Object?)accountAddress);
     }
 
-    public static bool operator !=(AccountAddress? left, AccountAddress? right)
+    public override int GetHashCode()
     {
-        return !Equals(left, right);
+        return ToString().GetHashCode();
     }
 }
```

Comment:
We should have `is_alias_of` function which checks if an address is an alias of another. Similarly we should have a function to generate the `n`-th alias.